### ISSUE TITLE
Client Encryption : Avoid blocking on encryption result when creating Transactional batch

### ISSUE THREAD

**Issue by anujtoshniwal** [Azure/azure-cosmos-dotnet-v3#1575]
https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1571

@abhijitpai
I wonder whether all these should be done in ExecuteAsync - customer is not expecting the serialization+encryption happening when they are constructing the batch request that the CreateItem call is for, and we're having to block on the Result of the async method. We may need to expose operations within the batch and classes such as ItemBatchOperation (and possibly its members such as OperationType or equivalent) as public for this based on suggested pattern - @kirankumarkolli let me know if you have alternate suggestions. One alternative is to have a parallel structure here to hold the operations, and then create the whole batch and execute it within ExecuteAsync.

@kirankumarkolli
That's a great point.
Its not getting prettier. None of the layering models are right fits here.

How about unblock CX (with assumption that its non-blocking) and then revisit?

**Comment by anujtoshniwal** [comment]
Duplicate of https://github.com/Azure/azure-cosmos-dotnet-v3/issues/2737
